drivers/clock_control/clock_control_rpi_pico.c

```diff
@@ -608,17 +661,26 @@ static int clock_control_rpi_pico_init(const struct device *dev)
 	/* Reset all function before clock configuring */
 	reset_block(~(RESETS_RESET_IO_QSPI_BITS | RESETS_RESET_PADS_QSPI_BITS |
 		      RESETS_RESET_PLL_USB_BITS | RESETS_RESET_USBCTRL_BITS |
-		      RESETS_RESET_PLL_USB_BITS | RESETS_RESET_SYSCFG_BITS |
-		      RESETS_RESET_PLL_SYS_BITS));
+		      RESETS_RESET_SYSCFG_BITS | RESETS_RESET_PLL_SYS_BITS));
 
 	unreset_block_wait(RESETS_RESET_BITS &
-			   ~(RESETS_RESET_ADC_BITS | RESETS_RESET_RTC_BITS |
+			   ~(RESETS_RESET_ADC_BITS |
+#if defined(RESETS_RESET_RTC_BITS)
+			     RESETS_RESET_RTC_BITS |
+#endif
+#if defined(RESETS_RESET_HSTX_BITS)
```

Comment:
Since these are independent situations, it is better not to connect them with the "else".
Also here, it is better to check the existence of the clock from the device tree.

Reply:
> it is better not to connect them with the "else".

What `else` are you referring to? Can you provide a line number?

> Also here, it is better to check the existence of the clock from the device tree.

I disagree, for the reason set out above.